From d51c3efb2dce70c4ae45c11139c5f04191e4cac5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Luk=C3=A1=C5=A1=20Hr=C3=A1zk=C3=BD?= <lhrazky@redhat.com>
Date: Wed, 25 Nov 2020 18:45:49 +0100
Subject: [PATCH] test_system_upgrade: Set installroot in the mocked cli

Fixes running tests on Fedora 31.

Sets the installroot to the default "/" in the mocked cli.base.conf. The
whole object is mocked and on Fedora 31 the mocked installroot object wasn't
resulting in anything the command could use. Since all the directories
are already changed to temporary locations, we don't need to set the
installroot to anything special.
---
 tests/test_system_upgrade.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/tests/test_system_upgrade.py b/tests/test_system_upgrade.py
index a6b1495..04f9ced 100644
--- a/tests/test_system_upgrade.py
+++ b/tests/test_system_upgrade.py
@@ -253,6 +253,7 @@ class CommandTestCaseBase(unittest.TestCase):
         self.datadir = tempfile.mkdtemp(prefix="system_upgrade_test_datadir-")
         system_upgrade.SystemUpgradeCommand.DATADIR = self.datadir
         self.cli = mock.MagicMock()
+        self.cli.base.conf.installroot = "/"
         self.command = system_upgrade.SystemUpgradeCommand(cli=self.cli)
         self.command.base.conf.cachedir = os.path.join(self.datadir, "cache")
         self.command.base.conf.destdir = None
@@ -403,7 +404,6 @@ class DownloadCommandTestCase(CommandTestCase):
         self.command.opts.repos_ed = []
         self.cli.demands.allow_erasing = "allow_erasing"
         self.command.base.conf.best = True
-        self.command.base.conf.installroot = self.datadir
         self.command.base.conf.releasever = "35"
         self.command.base.conf.gpgcheck = True
         self.command.opts.destdir = self.datadir
@@ -431,7 +431,6 @@ class DownloadCommandTestCase(CommandTestCase):
         self.command.opts.repos_ed = []
         self.cli.demands.allow_erasing = "allow_erasing"
         self.command.base.conf.best = True
-        self.command.base.conf.installroot = self.datadir
         self.command.base.conf.releasever = "35"
         self.command.base.conf.gpgcheck = True
         self.command.opts.destdir = self.datadir
-- 
2.28.0

